compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/selection/Toggleable.kt

```diff
@@ -302,8 +317,17 @@ private fun Modifier.toggleableImpl(
             }
         )
         .then(semantics).then(focusRequesterModifier)
+        .detectToggleFromKey()
         .indication(interactionSource, indication)
         .hoverable(enabled = enabled, interactionSource = interactionSource)
         .focusable(enabled = enabled, interactionSource = interactionSource)
         .then(gestures)
 }
+
+/**
+ * Whether the specified [KeyEvent] represents a user intent to perform a toggle.
+ * (eg. When you press Space on a focused checkbox, it should perform a toggle).
+ */
+@OptIn(ExperimentalComposeUiApi::class)
+internal val KeyEvent.isToggle: Boolean
+    get() = type == KeyEventType.KeyUp && key == Key.Spacebar
```

Comment:
The reason why it was expect/actual in that PR, is because `Key.Spacebar` is experimental and we can't use it as a cross-module dependency. It can be changed in the future, and we'll get UnresolvedMethodReference, if the user uses a newer version of `ui` and the old version of `foundation`.

Reply:
I see now. Then if we have expect/actual this PR has to be upstreamed, right?
And our common publication we'll need to reference the android build that contains the actual?

Reply:
Hmm.. It seems I'm wrong. it's not necessarily has to be upstreamed. The behaviour will be different, but everything will keep working.

Reply:
> this PR has to be upstreamed

Yes, behaviour will be different, and Android just won't react to Space, because the Jetpack Compose compiled binary doesn't have this feature.

But in terms of API it will be compatible. In terms of source code, IDEA will show our commonMain source code, when we go to the definition. That can add some confusion, and can mislead the debugger. If I am not mistaken, we can look at the true source code of Android target (`-foundation-android-1.2.0-source.jar`), but we need to do more clicks than usual.

Reply:
I also realised, that we need to use the same approach in the other our focus features (arrow keys in drop down menu)

Reply:
expcet/actual is back + added one in jsNativeMain too.